machine_test.go

```diff
@@ -121,55 +130,68 @@ fi
 	exitcode, _ := m.Run(command)
 
 	if exitcode != 0 {
-		t.Fatalf("Test for tmpfs mount on scratch failed with %d", exitcode)
+		suite.T().Fatalf("Test for tmpfs mount on scratch failed with %d", exitcode)
 	}
 }
 
-func TestSpawnMachine(t *testing.T) {
+func (suite *FakemachineTestSuite) TestSpawnMachine() {
 
 	if InMachine() {
-		t.Log("Running in the machine")
+		suite.T().Log("Running in the machine")
 		return
 	}
 
-	m := CreateMachine(t)
+	m := suite.CreateMachine()
 
+	// TODO error: "testing: warning: no tests to run"
 	exitcode, _ := m.RunInMachineWithArgs([]string{"-test.run TestSpawnMachine"})
 
 	if exitcode != 0 {
-		t.Fatalf("Test for respawning in the machine failed failed with %d", exitcode)
+		suite.T().Fatalf("Test for respawning in the machine failed failed with %d", exitcode)
 	}
 }
 
-func TestImageLabel(t *testing.T) {
+func (suite *FakemachineTestSuite) TestImageLabel() {
 	if InMachine() {
-		t.Log("Running in the machine")
+		suite.T().Log("Running in the machine")
 		devices := flag.Args()
-		assert.Equal(t, len(devices), 2, "Only expected two devices")
+		assert.Equal(suite.T(), len(devices), 2, "Only expected two devices")
 
 		autolabel := devices[0]
 		labeled := devices[1]
 
 		info, err := os.Stat(autolabel)
-		assert.Nil(t, err)
-		assert.Equal(t, info.Mode()&os.ModeType, os.ModeDevice, "Expected a device")
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), info.Mode()&os.ModeType, os.ModeDevice, "Expected a device")
 
 		info, err = os.Stat(labeled)
-		assert.Nil(t, err)
-		assert.Equal(t, info.Mode()&os.ModeType, os.ModeDevice, "Expected a device")
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), info.Mode()&os.ModeType, os.ModeDevice, "Expected a device")
 
 		return
 	}
 
-	m := CreateMachine(t)
+	m := suite.CreateMachine()
 	autolabel, err := m.CreateImage("test-autolabel.img", 1024*1024)
-	assert.Nil(t, err)
+	assert.Nil(suite.T(), err)
 
 	labeled, err := m.CreateImageWithLabel("test-labeled.img", 1024*1024, "test-labeled")
-	assert.Nil(t, err)
+	assert.Nil(suite.T(), err)
 
+	// TODO error: "testing: warning: no tests to run"
 	exitcode, _ := m.RunInMachineWithArgs([]string{"-test.run TestImageLabel", autolabel, labeled})
 	if exitcode != 0 {
-		t.Fatalf("Test for images in the machine failed failed with %d", exitcode)
+		suite.T().Fatalf("Test for images in the machine failed failed with %d", exitcode)
 	}
 }
+
+func TestFakemachineTestSuite(t *testing.T) {
+	// TODO run some tests outside of fakemachine first - finding modules etc?
+	// TODO test auto backend resolution
+	// TODO try to speed up the test by bringing up just one fakemachine per backend?
+
+	// test all backends
+	suite.Run(t, &FakemachineTestSuite{backend: "auto"})
```

Comment:
maybe have all commands which run inside a backend called `FakemachineBackendTestSuite` or similar to allow tests which don't need a backend to be ran ?